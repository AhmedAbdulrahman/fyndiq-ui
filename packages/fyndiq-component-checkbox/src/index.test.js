import React from 'react'
import { shallow, mount } from 'enzyme'

import Checkbox from './'

describe('fyndiq-component-checkbox', () => {
  test('should be invoquable without parameters', () => {
    const component = shallow(<Checkbox />)
    expect(component).toMatchSnapshot()
  })

  test('should have prop checked', () => {
    const component = shallow(<Checkbox checked />)
    expect(component).toMatchSnapshot()
  })

  test('should have prop disabled', () => {
    const component = shallow(<Checkbox disabled />)
    expect(component).toMatchSnapshot()
  })

  test('should have prop color', () => {
    const component = shallow(<Checkbox checked color="red" />)
    expect(component).toMatchSnapshot()
  })

  test('should have prop onToggle', () => {
    const component = shallow(<Checkbox onToggle={jest.fn()} />)
    expect(component).toMatchSnapshot()
  })

  test('should call onToggle when clicked on', () => {
    const toggleSpy = jest.fn()
    const component = shallow(<Checkbox onToggle={toggleSpy} />)
    component.simulate('click')

    expect(toggleSpy.mock.calls).toHaveLength(1)
    expect(toggleSpy.mock.calls[0][0]).toBe(true)

    component.simulate('click')
    expect(toggleSpy.mock.calls[1][0]).toBe(false)
  })

  test('should update its state when clicked on', () => {
    const component = shallow(<Checkbox onToggle={jest.fn()} />)
    component.simulate('click')
    expect(component).toMatchSnapshot()

    component.simulate('click')
    expect(component).toMatchSnapshot()
  })

  test('should be programatically checkable', () => {
    const component = mount(<Checkbox />)
    component.setProps({ checked: true })
    expect(component).toMatchSnapshot()
  })
})
