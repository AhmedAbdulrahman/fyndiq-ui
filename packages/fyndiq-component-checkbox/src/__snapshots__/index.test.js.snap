// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-checkbox should be invoquable without parameters 1`] = `
<label
  className="
          wrapper
          false
        "
>
  <input
    checked={false}
    className="
            checkbox
            false
            
          "
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <Checkmark
    className="checkmark"
    color="#7E7E7E"
    height={15}
    width={15}
  />
</label>
`;

exports[`fyndiq-component-checkbox should be programatically checkable 1`] = `
<Checkbox
  checked={true}
  className=""
  color="#7E7E7E"
  disabled={false}
  frame={false}
  onToggle={[Function]}
>
  <label
    className="
          wrapper
          false
        "
  >
    <input
      checked={true}
      className="
            checkbox
            false
            
          "
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <Checkmark
      className="checkmark"
      color="#7E7E7E"
      height={15}
      width={15}
    >
      <svg
        className="checkmark"
        fill="#7E7E7E"
        height={15}
        viewBox="0 0 50 50"
        width={15}
      >
        <path
          d="M16.8 42c-.4 0-1-.2-1.2-.5l-14-14.2c-.8-.7-.8-1.8 0-2.5.6-.7 1.7-.7 2.4 0l12.8 13 29-29c.7-.7 1.8-.7 2.5 0s.7 1.8 0 2.5L18 41.5c-.2.3-.7.5-1.2.5z"
        />
      </svg>
    </Checkmark>
  </label>
</Checkbox>
`;

exports[`fyndiq-component-checkbox should have prop checked 1`] = `
<label
  className="
          wrapper
          false
        "
>
  <input
    checked={true}
    className="
            checkbox
            false
            
          "
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <Checkmark
    className="checkmark"
    color="#7E7E7E"
    height={15}
    width={15}
  />
</label>
`;

exports[`fyndiq-component-checkbox should have prop color 1`] = `
<label
  className="
          wrapper
          false
        "
>
  <input
    checked={true}
    className="
            checkbox
            false
            
          "
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <Checkmark
    className="checkmark"
    color="red"
    height={15}
    width={15}
  />
</label>
`;

exports[`fyndiq-component-checkbox should have prop disabled 1`] = `
<label
  className="
          wrapper
          wrapperDisabled
        "
>
  <input
    checked={false}
    className="
            checkbox
            false
            
          "
    disabled={true}
    onChange={[Function]}
    type="checkbox"
  />
  <Checkmark
    className="checkmark"
    color="#7E7E7E"
    height={15}
    width={15}
  />
</label>
`;

exports[`fyndiq-component-checkbox should have prop onToggle 1`] = `
<label
  className="
          wrapper
          false
        "
>
  <input
    checked={false}
    className="
            checkbox
            false
            
          "
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <Checkmark
    className="checkmark"
    color="#7E7E7E"
    height={15}
    width={15}
  />
</label>
`;

exports[`fyndiq-component-checkbox should update its state when clicked on 1`] = `
<label
  className="
          wrapper
          false
        "
>
  <input
    checked={true}
    className="
            checkbox
            false
            
          "
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <Checkmark
    className="checkmark"
    color="#7E7E7E"
    height={15}
    width={15}
  />
</label>
`;

exports[`fyndiq-component-checkbox should update its state when clicked on 2`] = `
<label
  className="
          wrapper
          false
        "
>
  <input
    checked={false}
    className="
            checkbox
            false
            
          "
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <Checkmark
    className="checkmark"
    color="#7E7E7E"
    height={15}
    width={15}
  />
</label>
`;
